iroha/src/kura.rs

```diff
@@ -37,6 +39,24 @@ impl Kura {
         }
     }
 
+    /// `Kura` constructor with a [Genesis
+    /// Block](https://en.wikipedia.org/wiki/Blockchain#cite_note-hadc-21).
+    /// Kura will not be ready to work with before `init` method invocation.
+    pub fn with_genesis_block(
+        mode: Mode,
+        block_store_path: &Path,
+        block_sender: CommittedBlockSender,
+        genesis_block: ValidBlock,
+    ) -> Self {
+        Kura {
+            mode,
+            block_store: BlockStore::with_genesis_block(block_store_path, genesis_block),
+            block_sender,
+            merkle_tree: MerkleTree::new(),
+            blocks: Vec::new(),
+        }
```

Comment:
I think you forgot to notify `wsv` about the new block. We should send `genesis_block` through `block_sender` to `wsv`.

Reply:
@eadventurous it will be done during `init`